projects/packages/search/src/dashboard/components/pages/dashboard-page.jsx

```diff
@@ -144,6 +156,35 @@ export default function DashboardPage( { isLoading = false } ) {
 	);
 }
 
+const PlanInfo = props => {
+	const hasIndex = props.hasIndex;
+	const info = props.recordMeterInfo;
+	// Site Info
+	// TODO: Investigate why this isn't returning anything useful.
+	const siteTitle = useSelect( select => select( STORE_ID ).getSiteTitle() ) || 'your site';
+	// Plan Info data
+	const currentPlan = useSelect( select => select( STORE_ID ).getCurrentPlan() );
+	const currentUsage = useSelect( select => select( STORE_ID ).getCurrentUsage() );
+	const latestMonthRequests = useSelect( select => select( STORE_ID ).getLatestMonthRequests() );
+	const tierSlug = useSelect( select => select( STORE_ID ).getTierSlug() );
+	const planInfo = { currentPlan, currentUsage, latestMonthRequests, tierSlug };
+	return (
+		<>
+			{ ! hasIndex && <FirstRunSection siteTitle={ siteTitle } planInfo={ planInfo } /> }
+			{ hasIndex && <PlanUsageSection planInfo={ planInfo } /> }
+			{ hasIndex && (
+				<RecordMeter
+					postCount={ info.postCount }
+					postTypeBreakdown={ info.postTypeBreakdown }
+					tierMaximumRecords={ info.tierMaximumRecords }
+					lastIndexedDate={ info.lastIndexedDate }
+					postTypes={ info.postTypes }
+				/>
+			) }
```

Comment:
What's the benefit of nesting the `RecordMeter` component inside `PlanInfo`?

Reply:
I was thinking it made sense to separate the render logic for new and old layouts based on one test. So if we're looking at the new pricing, everything is contained in the `PlanInfo` section. I was also anticipating that the RecordMeter might move further down the hierarchy if we end up putting it behind a toggle.

I suppose we could keep it at the top level and add another check for 'first run' status though. Let me know if you have a preference!

Also, is it correct that we'll be removing the feature gate once the new pricing has shipped? Or after a safe amount of time to account for hiccups or rollbacks?

Reply:
No preference, just wanted to see if you had a particular reason 🙂

Yes, I anticipate we'll remove the feature gate once we've shipped the new pricing tiers 👍